runners/gradle-plugin/src/main/kotlin/org/jetbrains/dokka/gradle/internal/AbstractDokkaTaskExtensions.kt

```diff
@@ -0,0 +1,18 @@
+package org.jetbrains.dokka.gradle.internal
+
+import org.jetbrains.dokka.InternalDokkaApi
+import org.jetbrains.dokka.gradle.AbstractDokkaTask
+import org.jetbrains.dokka.toPrettyJsonString
+import org.jetbrains.dokka.DokkaConfiguration
+
+/**
+ * Serializes [DokkaConfiguration] of this [AbstractDokkaTask] as pretty json
+ *
+ * Should be used for debugging only, no guarantees are given for the support of this API.
+ *
+ * Better alternative should be introduced as part of [#2873](https://github.com/Kotlin/dokka/issues/2873).
+ */
+@InternalDokkaApi
+fun AbstractDokkaTask.buildJsonConfiguration(): String {
+    return this.buildDokkaConfiguration().toPrettyJsonString()
+}
```

Comment:
Did you consider adding a parameter, `pretty: Boolean`, and then conditionally returning `toPrettyJsonString()` or `toCompactJsonString()`? 

It's fine as it is, and I don't see a strong argument for returning compact JSON. I'm just curious.

Reply:
Good point, definitely won't hurt 👍 will add it